src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -124,18 +127,32 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
 
             throttle = new IndexThrottle();
             this.searcherFactory = new SearchFactory(engineConfig);
+            final Tuple<Long, Long> translogId; // nextTranslogId, currentTranslogId
             try {
                 writer = createWriter();
+                indexWriter = writer;
+                translogId = loadTranslogIds(writer, translog);
             } catch (IOException e) {
                 throw new EngineCreationFailureException(shardId, "failed to create engine", e);
             }
-            indexWriter = writer;
-            manager = createSearcherManager();
+            manager = createSearcherManager(translogId.v2());
             this.searcherManager = manager;
             this.mergeSchedulerFailureListener = new FailEngineOnMergeFailure();
             this.mergeSchedulerListener = new MergeSchedulerListener();
             this.mergeScheduler.addListener(mergeSchedulerListener);
             this.mergeScheduler.addFailureListener(mergeSchedulerFailureListener);
+            final TranslogRecoveryPerformer transformer = engineConfig.getTranslogRecoveryPerformer();
+            try {
+                if (translogId.v2() != translogId.v1() // if both are the same we don't have anything to replay
```

Comment:
I'd prefer to have translogId.v2() set to null, as opposed to be indentical to next. To me it would be clearer, but if you feel differently I'm OK with leaving as is.